TACC/Core-Portal
# task/FP-1311: Loading Icon for _common Button React Component

## Overview: ##
Enabled support of loading spinner over text in _common Button component.

## Related Jira tickets: ##

* [FP-1311](https://jira.tacc.utexas.edu/browse/FP-1311)

## Summary of Changes: ##
- Added styles to Button's CSS module to position spinner over text and reduce prevalence of text on loading state
- Added `isLoading` prop to Button component for devs to show/hide spinner and reduce prevalence of button text, as well as disable button simultaneously
- Added an example of button loading state on UI Patterns 
- Fixed `.as-link` outline color bug overlooked from [FP-545](https://github.com/TACC/Core-Portal/pull/584)

## Testing Steps: ##
1. Navigate to UI Patterns and scroll to the **Button** section
2. Click on button with text 'Click Me!'
3. Confirm that loading spinner appears after click, button text is behind spinner and is less visible, the button size does not change, and that the button cannot be clicked again while loading.
4. Confirm loading state attributes in previous step disappear (timeout is set to 3s)

**Button as Link UI**
1. On same page, confirm button with text 'Link' does not have a rectangular outline.

## UI Photos:

https://user-images.githubusercontent.com/43251554/156449657-ba4fc078-35e1-4a24-9487-bd6146950cff.mov



## Notes: ##
- Having the loading state be controlled by implementations of the button via a prop seemed like the simplest way to support this, but please let me know of any other ideas you may have that could handle this loading state management better!
- If you have any ideas for tests to write for this part of the component, please let me know.
